### PR TITLE
Switch to using urlobject library

### PR DESCRIPTION
**Pull request by singingwolfboy**
Using a dedicated library is nicer than copy-pasting bits of Tornado. [URLObject docs are here.](https://urlobject.readthedocs.io/en/latest/index.html)

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def test_client_authentication_url(api_client, api_url):
             ('state', ''),
             ('redirect_uri', '/redirect'),
             ('response_type', 'code'),
-            ('client_id', 'None'),
+            ('client_id', ''),
```
Comment (singingwolfboy) [inline tests/test_client.py]: I don't know if you consider this a bug or a feature, but it's a change to call out. The `url_concat()` function would transform `None` values into the string "None". URLObject is more intelligent than that.

File: nylas/client/util.py [hunk 2]
```diff
@@ -1,30 +0,0 @@
-from six.moves.urllib.parse import urlencode
```
Comment (mhahnenberg) [inline nylas/client/util.py]: Hooray! 🎉

File: tests/test_client.py (hunk 1 above)
Comment (mhahnenberg) [inline tests/test_client.py]: Hmm, this is technically a breaking change. I know it's sorta weird, but would it be possible to go back to the old behavior and change the earlier line in `authentication_url` to `'client_id': self.app_id or 'None',`?

File: nylas/client/util.py (hunk 2 above)
Comment (mhahnenberg) [inline nylas/client/util.py]: Hooray! 🎉

**Pull request merged**